### ISSUE TITLE
Core dump in TdiCompile when compiling <no-node>

### ISSUE THREAD

**Issue by GabrieleManduchi** [MDSplus/mdsplus#2025]
TdiCompile crashes when expression <no-node> is passed. This is an incorrect expression, so TdiCompile should report a bad status, not crash.  

The issue can be easily replicated in tdic
```
TDI>compile('<no-node>')
```
It crashes regardless a tree has been open before.

Unfortunaltely <no-node> is the output of TdiDecompile for a non existing tree node, so it can happen that this unfortunate string is then passed to TdiCompile.

**Comment by zack-vii** [comment]
Please always state the version? I get
```tdic
TDI> MdsShr->MdsRelease:T()
"alpha_release_7.105.0"
TDI> compile("")
*
TDI>
```

**Comment by zack-vii** [comment]
Side node "" should not be the deco of a nonexisting node.. Is it a nid?.. it should probably decompile like DTYPE_L and if it is a PATH it should just convert to DTYPE_T, no?

**Comment by zack-vii** [comment]
oh i did not mean to close it

**Comment by zack-vii** [comment]
can you run the two commands and post the response please
```tdi
MdsShr->MdsRelease:T()
compile('')
```

**Comment by GabrieleManduchi** [comment]
Sorry, my example was not correct (GitHub canceled the special characters).
The crash occurs when you passes the string no-node within angular brackets

**Comment by zack-vii** [comment]
oh i see

**Comment by GabrieleManduchi** [comment]
"\<no-node\>"

**Comment by GabrieleManduchi** [comment]
No way, I am not able to write in github...

**Comment by zack-vii** [comment]
you need to excape the \< and \> with \\ as \\\<

**Comment by GabrieleManduchi** [comment]
ok, thanks

**Comment by zack-vii** [comment]
hm well this is a bad decompile.. but also it is bad that it crashes

**Comment by zack-vii** [comment]
if you wrap your code in 
\`\`\` 
"\<no-node\>"
\`\`\`

it becomes 
```
"<no-node>"
```
without the need of the escaping

**Comment by zack-vii** [comment]
I will see what i can do

possibly we should
1. change the decompile to and comments /\*no-node\*/ which would process as * aka $MISSING (deco string must be length 12 due to the nature of the implementation)
2. fix the compile of \<verb-verb\> which would be a syntax error?

**Comment by GabrieleManduchi** [comment]
agree

**Comment by zack-vii** [comment]
Hi @GabrieleManduchi 

can you test/approve the PR, please?

**Comment by zack-vii** [comment]
fixed by #2027
